content/blog/jlink-to-produce-own-runtime/index.md

```diff
@@ -0,0 +1,142 @@
+`legacy-jre` target it is, as the name suggests, legacy functionality.  The
+new LTS version provided us with an opportunity to make a clean break and we
+have decided now is the time to no longer provide JREs since there are more
+efficient options available. It also simplifies the list of downloads we have
+available and reduces our testing overhead.  We will continue to produce
+them for the existing 8 and 11 versions to retain continuity for those using
+them.  The discussion on this happened in
+[this issue](https://github.com/adoptium/temurin-build/issues/2683).
+
+The good news is that it is very easy to produce your own runtime that looks
+comparable to a legacy JRE, and in many cases it can be smaller than an old
+legacy JRE would be!
+
+## Sounds good. So how do I create my own "JRE"?
+
+Creating your own runtime that is comparable to a JRE is simpler
+then you might think.  Firstly, download and extract our JDK image.  You can
+then use the
+[jlink](https://docs.oracle.com/en/java/javase/17/docs/specs/man/jlink.html)
+command to create your own runtime that will be smaller, yet provide
+equivalent functionality to the JRE.  Replace `jdk-17+35` in the examples
+below with the version of Java you are working with, and replace the forward
+slashes with the path separator on your platform (e.g.  `\` for Windows):
+
+```
+   jdk-17+35/bin/jlink --add-modules ALL-MODULE-PATH --output jdk-17+35-jre \
+      --strip-debug --no-man-pages --no-header-files --compress=2
+```
+
+Once you have done that, the contents of the `jdk-17+35-jre` will work in
+place of a JRE and be significantly smaller.
+
+## What are the disk space savings?
+
+It will vary by platform and the version of Java you are using, but using
+one example with the `jdk-17+35` release, the full JDK for one platform is
+about 312Mb on disk.  The jlinked runtime using the above command is about
+95Mb.
+
+## Is this identical to the legacy JRE?
+
+Not quite. By default there are some extra modules included via the
+`ALL-MODULE-PATH` list than would be included in a normal JRE. You can list
+the ones which your runtime supports with `java --list-modules`. At the time
+of writing, using the `jdk-17+35` release, the list of extra modules is as
+follows:
+
+```
+ jdk.attach
+ jdk.compiler
+ jdk.editpad
+ jdk.hotspot.agent
+ jdk.internal.ed
+ jdk.internal.jvmstat
+ jdk.internal.le
+ jdk.internal.opt
+ jdk.jartool
+ jdk.javadoc
+ jdk.jcmd
+ jdk.jconsole
+ jdk.jdeps
+ jdk.jdi
+ jdk.jlink
+ jdk.jpackage
+ jdk.jshell
+ jdk.jstatd
+ jdk.random
+ jdk.unsupported.desktop
+```
+
+Also, to be able to support some of those modules, you will still have tools
+like `javac`, `jlink` and others in the runtime. The `--add-modules` command accepts a comma seperated list of modules
+instead of the full list implied by `ALL-MODULE-PATH`. You can use a
+comma-separated set of modules from the `--list-modules` output to limit it
+further if desired, for example to include the list of extras above. Doing
```

Comment:
"... if desired. For example, to include the extra set of modules listed above."